### PR TITLE
Fix byte-compiler, checkdoc, package-lint warnings

### PR DESCRIPTION
**Pull request by progfolio**
Add containing group for fretboard defgroup.
Add URL, Keywords metadata.
Properly require symbol-overlay library.

### CODE REVIEW

File: fretboard.el [hunk 1]
```diff
@@ -31,20 +32,20 @@
 ;; To highlight semitones, prefix the key with # (yes, prefix!).
 
 ;;; Code:
+(require 'symbol-overlay)
 
 (defgroup fretboard nil
-  "A guitar fretboard visualization.")
+  "A guitar fretboard visualizer."
+  :group 'fretboard)
```
Comment (vifon) [inline fretboard.el]: Is such a recursive customization group a common thing?

File: fretboard.el [hunk 2]
```diff
@@ -31,20 +32,20 @@
...
 (defgroup fretboard-faces nil
   "Faces in `fretboard-mode'."
   :group 'fretboard)
 
 (defun fretboard-highlight-note ()
   "Highlight the note corresponding to the pressed key.
-
 The key is being read directly from the used keybinding."
   (interactive)
-  (require 'symbol-overlay)
```
Comment (vifon) [inline fretboard.el:47]: My intention was to load it only when using the highlighting functionality, but you're probably right it's overengineered.

Comment (progfolio) [reply fretboard.el:47]: The byte compiler does will not load the dependency during byte compilation otherwise.

File: fretboard.el (hunk 1 above)
Comment (progfolio) [reply fretboard.el]: It won't hurt anything. Technically you should add it to one of the default groups.
Applications is probably the appropriate choice. I've amended the commit to reflect that.

File: fretboard.el (hunk 2 above)
Comment (vifon) [reply fretboard.el:47]: Yes, I'm aware. I meant that this little added amount of lazy loading is not worth having byte compilation warnings. Or complicating the code with `eval-when-compile`.

**Comment by vifon** [comment]
Merged! Thank you!

**Pull request merged**